### PR TITLE
[@types/onfleet_node-onfleet] Add new resource method types

### PR DESCRIPTION
**Pull request by marcobeltempo**
# Description

**Docs**
- [node-onfleet supported crud operations](https://github.com/onfleet/node-onfleet#supported-crud-operations)
- [Onfleet API docs](https://docs.onfleet.com/reference#introduction) 

This PR includes the following type changes/updates for the https://github.com/onfleet/node-onfleet project:

**Onfleet class**

- updated  signature
    - add "timeout" prop type
    - add "BottleneckOptions" prop type
- added "administrators" resource key

**Administrators**

- Added "matchMetadata"

**Destinations**

- Added "matchMetadata"

**Hubs**

- Added "create" method
- Added "update" method

**Recipients**

- Added "matchMetadata" method

**Tasks**

- Added "matchMetadata" method

**Team**

- Added "autoDispatch" method
- Added "getWorkerEta" method

**Workers**

- Added "matchMetadata" method

**Misc**

- bump minor version to 1.2
- added tests for new method types
- prettier fixes

Please fill in this template.

- [x] Use a meaningful title for the pull request. Include the name of the package modified.
- [x] Test the change in your own code. (Compile and run.)
- [x] [Add or edit tests](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#my-package-teststs) to reflect the change.
- [x] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [x] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [x] [Run `npm test <package to test>`](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#running-tests).

Select one of these and delete the others:

If changing an existing definition:
- [x] Provide a URL to documentation or source code which provides context for the suggested changes: <<url here>>
- [x] If this PR brings the type definitions up to date with a new version of the JS library, update the version number in the header.

### CODE REVIEW

File: types/onfleet__node-onfleet/onfleet.d.ts [hunk 1]
```diff
@@ -9,31 +9,43 @@ import Teams = require('./Resources/Teams');
...
+        headers: {
+            'Content-Type': string;
+            'User-Agent': string;
+            Authorization: string;
+        };
+    };
 
-  constructor(api_key: string);
-  verifyKey(): Promise<boolean>;
+    constructor(api_key: string, timeout?: number, bottleneckOptions?: BottleneckOptions);
```
Comment (efgallegos) [inline types/onfleet__node-onfleet/onfleet.d.ts:35]: @marcobeltempo - Can we add also this param `baseURL?: string` to the constructor. I just merged a this [PR](https://github.com/onfleet/node-onfleet/pull/46) to master in order to include this new param in @onfleet/node-onfleet. Thanks!

**Review by orta — APPROVED** [review]
Cool, looks generally reasonable to me

File: types/onfleet__node-onfleet/onfleet.d.ts (hunk 1 above)
Comment (marcobeltempo) [reply types/onfleet__node-onfleet/onfleet.d.ts]: Thanks for the heads up @efgallegos. I've gone ahead and added the optional `baseURL` param [(fa2c511)](https://github.com/DefinitelyTyped/DefinitelyTyped/pull/55000/commits/fa2c5114b5e695199c3b87c06b5616748c4d8905)

**Review by orta — APPROVED** [review]

**Comment by marcobeltempo** [comment]
Ready to merge

**Pull request merged**